propertyestimator/protocols/coordinates.py

```diff
@@ -13,96 +13,76 @@
+from propertyestimator.workflow.decorators import protocol_input, protocol_output, UNDEFINED
 from propertyestimator.workflow.plugins import register_calculation_protocol
 from propertyestimator.workflow.protocols import BaseProtocol
 
 
 @register_calculation_protocol()
 class BuildCoordinatesPackmol(BaseProtocol):
-    """Creates a set of 3D coordinates with a specified composition.
-
-    Notes
-    -----
-    The coordinates are created using packmol.
+    """Creates a set of 3D coordinates with a specified composition
+    using the PACKMOL package.
     """
 
-    @protocol_input(int)
-    def max_molecules(self):
-        """The maximum number of molecules to be added to the system."""
-        pass
-
-    @protocol_input(unit.Quantity)
-    def mass_density(self):
-        """The target density of the created system."""
-        pass
-
-    @protocol_input(list)
-    def box_aspect_ratio(self):
-        """The aspect ratio of the simulation box. The default is [1.0, 1.0, 1.0],
-        i.e a cubic box."""
-        return self._box_aspect_ratio
-
-    @protocol_input(Substance)
-    def substance(self):
-        """The composition of the system to build."""
-        pass
-
-    @protocol_input(bool)
-    def verbose_packmol(self):
-        """If True, packmol will be allowed to log verbose information to the logger,
-        and any working packmol files will be retained."""
-        pass
-
-    @protocol_input(bool)
-    def retain_packmol_files(self):
-        """If True, packmol will not delete all of the temporary files it creates
-        while building the coordinates."""
-        pass
-
-    @protocol_output(int)
-    def output_number_of_molecules(self):
-        """The number of molecules in the created system. This may be less than
-        maximum requested due to rounding of mole fractions.
-        """
-        pass
-
-    @protocol_output(Substance)
-    def output_substance(self):
-        """The substance which was built by packmol. This may differ from the input
-        substance for system containing two or more components due to rounding of
-        mole fractions. The mole fractions provided by this output should always be
-        used when weighting values by a mole fraction.
-        """
-        pass
-
-    @protocol_output(str)
-    def coordinate_file_path(self):
-        """The file path to the created PDB coordinate file."""
-        pass
+    max_molecules = protocol_input(
+        docstring='The maximum number of molecules to be added to the system.',
+        type_hint=int,
+        default_value=1000
+    )
+    mass_density = protocol_input(
+        docstring='The target density of the created system.',
+        type_hint=unit.Quantity,
+        default_value=0.95 * unit.grams / unit.milliliters
+    )
+
+    box_aspect_ratio = protocol_input(
+        docstring='The aspect ratio of the simulation box.',
+        type_hint=list,
+        default_value=[1.0, 1.0, 1.0]
```

Comment:
I'm not 100% sure, but it's not clear whether an object which uses this value would use a _copy_ of this list, or _the list itself_ (in which case modifying one instance of the class could affect another)